### PR TITLE
Provide JSDoc comments for flag properties

### PR DESCRIPTION
**Pull request by tommy-mitchell**
Adds JSDoc comments to each property of the type `Flag` so that Intellisense is provided when creating a flag:

image

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -16,11 +16,75 @@ Callback function to determine if a flag is required during runtime.
...
+	*/
 	readonly isMultiple?: IsMultiple;
+
+	/**
+	Other names for the flag.
+
+	@example
+	```
+	unicorn: {
+		aliases: ['unicorns', 'u']
```
Comment (sindresorhus) [inline index.d.ts:84]: It should not contain `u` as users should use the shortFlag option for that.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by tommy-mitchell** [comment]
I also added missing info to the flags JSDoc that was in the readme, and reordered the options to group similar concepts.

**Comment by tommy-mitchell** [comment]
I'd like to help get #211 merged before the next release goes out. Should I open a new PR branched from it, or should it be reopened and I make PRs to its feature branch?

**Comment by sindresorhus** [comment]
> Should I open a new PR branched from it,

New PR.

**Comment by tommy-mitchell** [comment]
@sindresorhus unless you'd want to get #229 or #221 merged before the next release, you're good to put it out. Adding subcommands will probably be better with some general refactoring (separating utilities/steps into separate files, doing the same with tests/some fixtures, etc). I think that would be a good time to work on things like bundling and making errors consistent as well. I'll open a draft PR for that soon.

**Comment by sindresorhus** [comment]
There's no rush in getting out a new version. It's going to be a major bump.

**Pull request merged**